### DISCUSSION TITLE
404 Error on manual route navigation/refresh on web site

### DISCUSSION THREAD

**Discussion by roger-cprops** [quasarframework/quasar#15709] [General - Components / Directives / etc] [answered]
I've deployed an SPA created with Quasar v2 to an Azure app service and also GitHub.io and am getting a strange behavior I don't see when running in dev mode.

Programatic navigation works to any page fine using vue router (e.g. router.push({name: "clients"). 

However if I manually enter the url (e.g. https://mysite.github.io/clients) or hit refresh while on any page other than root, I get a 404 error.

Github.io displays this message: "The site configured at this address does not contain the requested file."

Any help would be appreciated

**Comment by suleimanas** [comment]
you need a fallback htmlfile, in github i think it's **/404.html**

**Reply by roger-cprops** [reply]
Hi, thanks for the answer but I don't think it addresses my issue.

I'll try to rephrase my question.

I expect that if I hit refresh on a page or type in the url manually that it will stay on/go to that page.  Why isn't it working this way and what do I need to do ti fix it?  Again, programmatic navigation works fine.

**Reply by suleimanas** [reply, marked as answer]
Hello, i understood your question and believe me it is addressing your issue.
the issue is not in your SPA it's a general static hosting issue, 
just copy your **/index.html** to **/404.html** and try it for yourself

**Reply by roger-cprops** [reply]
Thanks so much. That fixed it on GitHub.

**Reply by rstoenescu** [reply]
This works but I want to expand on an explanation.

It matters what Vue Router you choose to use, based on your hosting capabilities. For static hosting, one should go with "hash" and not "history".. unless you can configure your 404 to use /index.html too.

By using "history" mode your URL path will change when navigating between pages, so naturally by refreshing the page the browser will try to fetch the current URL again -- leading into 404s. So you must configure your hosting to point to /index.html regardless of the URL path.
Alternatively, "hash" mode keeps the current path (which will always be valid on refresh) and only changes the hash part when navigating.
